### ISSUE TITLE
Not able to interact with the Hammer.js tap event

### ISSUE THREAD

**Issue by jvegarag** [cypress-io/cypress#4286]
### Context

We are trying to perform automated testing for a web application based on Angular 5. Due to the web app must be accessible through mobile and tablet devices, we use [Hammer.js](https://hammerjs.github.io/) in order to be able to handle some gesture-based events.

### Current behavior:

Cypress is not able to click on a button that is listening the Hammer.js "tap" event. When clicking, the handler is not executed. On the other hand, the user has no problem to interact by using the mouse click event.

### Desired behavior:

The behaviour should be the same as the expected when the click event is used otherwise we cannot use Cypress in order to "simulate" the user interaction with the UI.

We successfully tested with other automation tools based on Selenium webdriver but we are interested on migrating our automated tests to Cypress. Nevertheless, this issue is blocking for us.

### Versions
3.3.1

**Comment by jennifer-shehane** [comment]
Testing clicking on hammer.js, I can see that the click doesn't really perform the reaction that it should.

my code

```js
it('click Hammer.js', () => {
  cy.visit('https://hammerjs.github.io/')
  cy.get('#hitarea')
    .should('have.attr', 'style').and('include', 'rotate3d(0, 0, 0, 0deg);')
  cy.get('#hitarea')
    .click()
    .should('have.attr', 'style').and('include', 'rotate3d(0, 0, 0, 25deg);')
})
```

I determined this by turning on the Event Listener Breakpoints for Mouse events on both the regular browser while clicking and the Cypress browser while running my test. The click event is being hit in the same piece of code, but the animation is not occuring. 

There are no `touch` events that are triggered within Hammer.js during mouse click when used in regular browser, so we don't need to worry about this.

<img width="596" alt="Screen Shot 2019-05-23 at 3 09 38 PM" src="https://user-images.githubusercontent.com/1271364/58238189-cd8caf80-7d6c-11e9-9f72-cab87d214308.png">

I actually think that hammer.js is detecting that this is a synthetic event - so changing it's behavior based on that? Honestly, I'm not sure. I'm sure native events will solve this issue though. https://github.com/cypress-io/cypress/issues/311

**Comment by jvegarag** [comment]
Hi @jennifer-shehane ,

I finally managed to perform native clicks + D&D by connecting Cypress to chrome devtools protocol via Puppeteer. I developed a POC plugin to do so: https://www.npmjs.com/package/cypress-puppeteer-plugin

I know I have lost the support of Electron browser in headless mode but that's not a problem for my project in that phase.

I would like to know if your approach to implement native events and other know limitations will be similar to that and if it's far in the future.

Thanks.

**Comment by jennifer-shehane** [comment]
@jvegarag Follow our Native Events issue here - https://github.com/cypress-io/cypress/issues/311 - all the work is open source, currently being worked on by @Bkucera mostly.

**Comment by sainthkh** [comment]
It seems that the bug has been fixed in Hammerjs v2.0.8 and Cypress 4.11.0. 

Below is the code I tested:

`index.html`

```html
<html>
  <head>
    <style>
      #hitarea {
        background: silver;
        height: 300px;
        text-align: center;
        font: 30px/300px Helvetica, Arial, sans-serif;
      }
    </style>
    <script src="https://hammerjs.github.io/dist/hammer.js"></script>
  </head>
  <body>
    <div id="hitarea"></div>
    <script>
      var myElement = document.getElementById('hitarea');

      // We create a manager object, which is the same as Hammer(), but without the presetted recognizers. 
      var mc = new Hammer.Manager(myElement);


      // Tap recognizer with minimal 2 taps
      mc.add( new Hammer.Tap({ event: 'doubletap', taps: 2 }) );
      // Single tap recognizer
      mc.add( new Hammer.Tap({ event: 'singletap' }) );


      // we want to recognize this simulatenous, so a quadrupletap will be detected even while a tap has been recognized.
      mc.get('doubletap').recognizeWith('singletap');
      // we only want to trigger a tap, when we don't have detected a doubletap
      mc.get('singletap').requireFailure('doubletap');


      mc.on("singletap doubletap", function(ev) {
          myElement.textContent += ev.type +" ";
      });
    </script>
  </body>
</html>
```

`test.js`

```js
it('click Hammer.js', () => {
  cy.visit('/fixtures/issue-4286.html')
  cy.get('#hitarea')
  .click()
  .should('have.text', 'singletap ')
  .dblclick()
  .should('have.text', 'singletap doubletap ')
})
```

I borrowed [the official example code about the tap event](https://codepen.io/jtangelder/pen/pBuIw).

**Comment by sainthkh** [comment]
If the same problem happens in your code, please leave a comment or open a new issue.
